packages/components/app/styles/components/copy/snippet.scss

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// COPY > SNIPPET
+//
+@use "../../mixins/focus-ring" as *;
+
+.hds-copy-snippet {
+  display: flex;
+  gap: 0.5rem;
+  align-items: center;
+  align-self: stretch;
+  padding: 0.375rem 0.25rem;
+  font-size: var(--token-typography-code-100-font-size);
+  font-family: var(--token-typography-code-100-font-family);
+  line-height: var(--token-typography-code-100-line-height);
+  text-align: left;
+  border: 1px solid transparent;
+  border-radius: 0.3125rem;
+}
+
+.hds-copy-snippet--color-primary {
+  color: var(--token-color-foreground-action);
+  background-color: transparent;
+
+  &:hover,
+  &.mock-hover {
+    background-color: var(--token-color-surface-interactive);
+    border-color: var(--token-color-border-strong);
+  }
+
+  &:active,
+  &.mock-active {
+    color: var(--token-color-foreground-action-active);
+    background-color: var(--token-color-surface-interactive-active);
+    border-color: var(--token-color-border-strong);
+  }
+
+  &:focus,
+  &.mock-focus {
+    @include hds-focus-ring-with-pseudo-element();
+  }
+}
+
+.hds-copy-snippet--color-secondary {
+  color: var(--token-color-foreground-primary);
+  background-color: transparent;
+
+  &:hover,
+  &.mock-hover {
+    color: var(--token-color-foreground-primary);
+    background-color: var(--token-color-surface-interactive);
+    border-color: var(--token-color-border-strong);
+  }
+
+  &:active,
+  &.mock-active {
+    color: var(--token-color-foreground-primary);
+    background-color: var(--token-color-surface-interactive-active);
+    border-color: var(--token-color-border-strong);
+    border-radius: 0.3125rem;
+  }
+
+  &:focus,
+  &.mock-focus {
+    @include hds-focus-ring-with-pseudo-element();
+  }
+
+  .hds-copy-snippet__icon {
+    color: var(--token-color-foreground-action);
+
+    &:hover {
+      color: var(--token-color-foreground-action-hover);
+    }
+
+    &:active {
+      color: var(--token-color-foreground-action-active);
+    }
+
+    &:focus {
+      color: var(--token-color-foreground-action);
+    }
+  }
+}
+
+.hds-copy-snippet--success {
+  background-color: var(--token-color-surface-interactive);
+
+  .hds-copy-snippet__icon {
+    color: var(--token-color-foreground-success);
```

Comment:
Same question as the `CopyButton`, should this be `--token-color-foreground-success-on-surface`?